mu-plugins/blocks/navigation/index.php

```diff
@@ -0,0 +1,142 @@
+					apply_filters( 'wporg_block_navigation_menus', array() )
+				);
+				wp_enqueue_script( $editor_script_handle );
+			}
+		}
+	);
+
+	// Hide the menu selection when a dynamic menu is selected.
+	add_action(
+		'admin_print_styles',
+		function() {
+			global $hook_suffix;
+			if ( ! in_array( $hook_suffix, array( 'post.php', 'post-new.php' ) ) ) {
+				return;
+			}
+			echo '<style>.wporg-nav-hide-next-panel + .components-panel__body { display: none; }</style>';
+		}
+	);
+}
+
+/**
+ * Update the inner blocks (menu items) in a navigation block.
+ *
+ * @param WP_Block_List $inner_blocks
+ */
+function update_navigation_items( $inner_blocks ) {
+	// This contains the parent block, so that we can check that we're in the correct Navigation.
+	$block = WP_Block_Supports::$block_to_render;
+	if (
+		isset( $block['blockName'] ) &&
+		'core/navigation' === $block['blockName'] &&
+		isset( $block['attrs']['menuSlug'] ) &&
+		$block['attrs']['menuSlug']
+	) {
+		$menu_content = get_menu_content( $block['attrs']['menuSlug'] );
+		$parsed_blocks = parse_blocks( $menu_content );
+		$compacted_blocks = block_core_navigation_filter_out_empty_blocks( $parsed_blocks );
+		$inner_blocks = new WP_Block_List( $compacted_blocks, $block['attrs'] );
+	}
+	return $inner_blocks;
+}
+
+/**
+ * Get the navigation menu content as blocks.
+ *
+ * @param string $menu_slug
+ *
+ * @return string Menu items in block syntax.
+ */
+function get_menu_content( $menu_slug ) {
+	$menu_items = get_menu_items( $menu_slug );
+	if ( ! $menu_items ) {
+		return '';
+	}
+
+	$menu_content = '';
+	foreach ( $menu_items as $item ) {
+		$block_code = '<!-- wp:navigation-link {"label":"%1$s","url":"%2$s","kind":"custom"} /-->';
+
+		// If this is a relative link, convert it to absolute and try to find
+		// the corresponding ID, so that the `current` attributes are used.
+		if ( str_starts_with( $item['url'], '/' ) ) {
+			$page_obj = get_page_by_path( $item['url'] );
+			$item['url'] = home_url( $item['url'] );
+			if ( $page_obj ) {
+				// A page was found, so use the post-type link.
+				$block_code = '<!-- wp:navigation-link {"label":"%1$s","url":"%2$s","kind":"post-type","id":"%3$s"} /-->';
+				$item['id'] = $page_obj->ID;
+			}
+		}
+
+		$menu_content .= sprintf(
+			$block_code,
+			esc_html( $item['label'] ),
+			esc_url( $item['url'], ),
+			isset( $item['id'] ) ? intval( $item['id'] ) : ''
+		);
+	}
+
+	return $menu_content;
+}
+
+/**
+ * Get a list of menu items for a given menu slug.
+ *
+ * This is used to build up the navigation menu. Relative links should
+ * correspond to pages on the site, while absolute URLs can be used to
+ * navigate off-site.
+ *
+ * @param string $menu_slug
+ *
+ * @return array|boolean List of menu items with label, url, or false if the
+ *                       slug does not match a navigation list.
+ */
+function get_menu_items( $menu_slug ) {
+	$menus = apply_filters( 'wporg_block_navigation_menus', array() );
```

Comment:
Could you add some documentation to this filter to describe the expected format for `$menus`? Like [this for the Query Filter filter](https://github.com/WordPress/wporg-mu-plugins/blob/c79d373ffdb91cd14cc12f08e23173fa872921e8/mu-plugins/blocks/query-filter/render.php#L6-L27) (I also referenced [the core docs for that](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#4-hooks-actions-and-filters)).

Reply:
Done in https://github.com/WordPress/wporg-mu-plugins/pull/443/commits/6ca4d6844c6f793d82da0a8318724a90482dde60